src/02_spec/07_modules/dem_uart/systemif.rst

```diff
@@ -0,0 +1,63 @@
+System Interface
+----------------
+
+The following signals are required from a bus in order to connect to the DEM.
+
++-----------------+----------+---------------+-------------------------------------------------+
+| Signal          | Width    | Direction     | Description                                     |
++=================+==========+===============+=================================================+
+| ``bus_req``     |    1     | CPU->DEM      | ``1`` indicates an active request from the CPU  |
++-----------------+----------+---------------+-------------------------------------------------+
+| ``bus_addr``    |    3     | CPU->DEM      | Address to be used with write/read operation    |
++-----------------+----------+---------------+-------------------------------------------------+
+| ``bus_write``   |    1     | CPU->DEM      | ``1`` indicates a register write request        |
++-----------------+----------+---------------+-------------------------------------------------+
+| ``bus_wdata``   |    8     | CPU->DEM      | Data to be written into the register            |
++-----------------+----------+---------------+-------------------------------------------------+
+| ``bus_ack``     |    1     | DEM->CPU      | Acknowledge last request                        |
++-----------------+----------+---------------+-------------------------------------------------+
+| ``bus_rdata``   |    8     | DEM->CPU      | Data that was read from the register            |
++-----------------+----------+---------------+-------------------------------------------------+
+
+A new request to this device is made by asserting ``bus_req``, ``bus_write`` indicates whether it
+is a read or write request. A valid register address is presented on ``bus_addr``.
+Depending on ``bus_write`` the current content of the register is either written to ``bus_rdata``,
+or overwritten with ``bus_wdata``. Finally ``bus_ack`` is asserted by the module after completing the
+request to properly end the bus cycle. For a list of valid register addresses see :ref:`uart_registers`
+
+------------
+Wishbone Bus
+------------
+The provided module ``osd_dem_uart_wb`` may be used to directly connect to a WISHBONE bus.
```

Comment:
Please do not document the implementation here, but write a specification. I.e. osd_dem_uart_wb is the name in your specific implementation; leave that out. Instead, describe something like "if a WB interface is provided, these signals must be present"